exports/internal.go

```diff
@@ -159,9 +160,14 @@ func (i *Internal) PostUpload(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	w.WriteHeader(http.StatusAccepted)
+	r.Body = http.MaxBytesReader(w, r.Body, int64(i.Cfg.MaxPayloadSize))
 
 	if err := i.Compressor.CreateObject(r.Context(), logger, i.DB, r.Body, params.Application, params.ResourceUUID, payload); err != nil {
+		if errors.As(err, &maxBytesError) {
+			w.WriteHeader(http.StatusRequestEntityTooLarge)
+			Logerr(fmt.Fprintf(w, "payload is too large, max size: %dMB", i.Cfg.MaxPayloadSize))
+		}
+
```

Comment:
We might need to return right here.  I'd need more time to dig into the code to decide for sure.  Can you see what the pros/cons are of returning here?